### PR TITLE
Fix Concurrency 3.1 Test Suite

### PR DESCRIPTION
**Pull request by KyleAure**
When attempting to run the `com.ibm.ws.concurrent_fat_jakarta` test suite against Concurrency 3.0 and Java 11 the following will occur:

1. We will compile the concurrency 3.1 bundle using Java 17
2. We will compile our test classes using Java 11
3. At runtime our test infrastructure (running on java 11) will attempt to load classes (such as the servlet class, or packages/classes passed to shrinkwrap).
4. Our test infrastructure will fail to load these classes since we only have the concurrency 3.1 API available and it was compiled at Java 17.

We will need to separate out the concurrency 3.1 tests into their own test bucket to resolve this issue.

### CODE REVIEW

File: dev/com.ibm.ws.concurrent_fat_jakarta_11/fat/src/test/jakarta/concurrency/Concurrency31Test.java [hunk 1]
```diff
@@ -33,7 +33,7 @@
 import test.jakarta.concurrency31.web.Concurrency31TestServlet;
 
 @RunWith(FATRunner.class)
-@MinimumJavaLevel(javaLevel = 21)
+@MinimumJavaLevel(javaLevel = 17)
```
Comment (njr-11) [inline dev/com.ibm.ws.concurrent_fat_jakarta_11/fat/src/test/jakarta/concurrency/Concurrency31Test.java]: I don't think some of the tests in this bucket will pass on Java SE 17 without updates.

Comment (KyleAure) [reply dev/com.ibm.ws.concurrent_fat_jakarta_11/fat/src/test/jakarta/concurrency/Concurrency31Test.java]: Yeah, I was testing locally using java 21.  I reverted this change.

**Comment by KyleAure** [comment]
#build

**Review by njr-11 — APPROVED** [review]

**Comment by KyleAure** [comment]
#libby - recheck copyright headers

**Pull request merged**